### ISSUE TITLE
Changing quantity inside the Mini cart block does not disable the buttons

### ISSUE THREAD

**Issue by borislavfil** [woocommerce/woocommerce#42480]
Changing quantity inside the Mini cart block does not disable the buttons during calculation. This could result in different than expected quantity at checkout. 
I think it would be best to disable the buttons and possibly show a spinner on the checkout button similar to the regular cart block

**Comment by mikejolley** [comment]
Are you referring to using mini-cart *on* the checkout page?

**Comment by borislavfil** [comment]
No, I mean that you can change the quantity and immediately click the Proceed to checkout button, before the first request goes through. In the cart component there is code which disables the button and activates a spinner in this case. It seems to be missing for the mini-cart.

**Comment by borislavfil** [comment]
I can make a pull request about this, but I am not sure if both buttons should get a spinner or just one of them with some logic since the availability of each button seems to be conditional.

**Comment by mikejolley** [comment]
Thanks for clarifying this. This may need design input but I think disabling the button would be a decent stop gap solution. CC'ing @Aljullu for feedback.

**Comment by Aljullu** [comment]
The proposal makes sense to me. I agree that both buttons should be disabled when doing calculations. I don't have a strong opinion design-wise, but I would keep it consistent with the Cart block (cc @vivialice).

@borislavfil feel free to ping me if you end up creating a PR. :slightly_smiling_face:

**Comment by albarin** [comment]
I might be missing something, but I think the button spinner on the `Cart` block only shows _after_ clicking the `Proceed to checkout` button, not when the product quantity is changed.

**Comment by borislavfil** [comment]
You are right. There is no spinner, but the button becomes disabled. Maybe this is enough...

**Comment by albarin** [comment]
I have created [this PR](https://github.com/woocommerce/woocommerce-blocks/pull/8715) as a proof of concept, this could be one solution. The buttons get disabled and show the spinner. Any feedback is appreciated. Thanks! 😄

**Comment by ralucaStan** [comment]
There is now a disabled state for the `Proceed to checkout` button on the Cart block when the shopper increases the quantity. I think the same should apply for the Mini Cart for consistency reasons. Might be worth looking at this PR by @mikejolley https://github.com/woocommerce/woocommerce-blocks/pull/10472

<img width="496" alt="Screenshot 2023-08-28 at 14 35 31" src="https://github.com/woocommerce/woocommerce-blocks/assets/1628454/75854573-cbce-4b3f-9be4-8107b8ca833c">

**Comment by pmcpinto** [comment]
> There is now a disabled state for the Proceed to checkout button on the Cart block when the shopper increases the quantity. I think the same should apply for the Mini Cart for consistency reasons. 

This is a good solution to this problem. I'll remove the needs:design label

**Comment by ralucaStan** [comment]
This fix should also be applied to the cart block. there the button is disabled, but visually there is no change.

**Comment by pierorocca** [comment]
Hi @borislavfil - thanks for the report.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

I've taken a look and marked this ready for the engineering team to pick up.

What I'm seeing: The mini-cart checkout button block still does not read checkout calculating state or disable itself during quantity recalculation, while the full cart proceed-to-checkout block does both and shows a spinner. The gap described in the issue is still visible in current code.

**Comment by j111q** [comment]
just stumbled on this issue, highlighting it to you @yjailin if you haven't seen it and have thoughts 🤔

**Comment by yjailin** [comment]
I would disable the Checkout button while the quantity updates, but also use the skeleton loading on the cart Subtotal. View cart stays enabled, it recalculates anyway. WDYT?

<img width="2064" height="1330" alt="Image" src="https://github.com/user-attachments/assets/56feaa78-d897-4918-a47f-78805c9f2b71" />

fpl1YUEIW7suBFO09qymA2-fi-9168_168307
